### ISSUE TITLE
janus.js should be NPM installable

### ISSUE THREAD

**Issue by cmacq2** [meetecho/janus-gateway#793]
Ideally janus.js and its JS dependencies would be installable via NPM, to better integrate with modern frontend workflows. 

That would require developing a NPM package, and it may be a good idea to move the janus.js code to a separate repository. Doing so could also enable us to tackle the forked adapter.js more cleanly, and remove the need to reference external files in the demos entirely. The demos could be part of a `janus-demos` NPM package which could pull in `janus.js` and other deps via NPM as normal. Finally, it could simplify the existing issues surrounding proper licensing and attribution of the code as well: as it is, the HTML/JS code sits somewhat awkwardly alongside the main C code base of Janus in this one.

Agree? Disagree?

**Comment by mirkobrankovic** [comment]
Uuu i agree completely :)

**Comment by lminiero** [comment]
Discussed this more than once. I don't like the idea of making janus.js any more complicated than it is and I can handle. I'm not a great JavaScript developer, and I have to maintain this stuff too. Changes too big or complex for me to be able to handle are going to take time from the server stuff, and I really don't want that to happen. Not sure if #792 can be seen as a step in your direction, but I told @thehunmonkgroup pretty much the same things.

If you do need anything more complex and flexible, there are [good JavaScript stacks](https://janus.conf.meetecho.com/docs/resources) to start from, so please don't make my life way harder than it already is... :smile:

**Comment by cmacq2** [comment]
Just to be clear: I am not asking to node-ify it. I'm asking to provide a simple NPM package wrapper around existing code so people can pull in Janus and get the maintained & up to date support for Janus REST API directly via NPM.

**Comment by thehunmonkgroup** [comment]
@cmacq2 I'm handling automatically keeping up to date via server build scripts, and, that's certainly a higher bar than 'npm install' ;)

@lminiero another perspective to consider: this proposal and #792 both have the advantage of making Janus more accessible/easier to install. I like the idea of removing barriers :)

**Comment by lminiero** [comment]
@cmacq2 good points, I'll definitely keep this in mind.

**Comment by zoobot** [comment]
Are you all saying that janus.js and dependencies are a js version of janus? Which files? Can I just use it with a node server?

I've been trying to get Janus installed for about an hour and had a ton of errors then failed on Raspbian. 

Would love an npm version of it, if it works!!

**Comment by lminiero** [comment]
There is no npm version of janus.js. If you want to control Janus from node.js, try one of the third-party libraries available [here](https://janus.conf.meetecho.com/docs/resources).

**Comment by SaqlainGardezi** [comment]
Agree. I am a node developer and even through documentation on janus site I am unable to get how can I integrate it into my Application. I have a  window machine

**Comment by lminiero** [comment]
Closing as we don't have any plan to make janus.js NPM installable. I'll consider merging the effort Chad proposed on UMD, but nothing more than that I'm afraid. We discussed having a separate project for the demos, but discarded that idea as well, as it would be hard to sync PRs (think of new features/plugins that add new demos or change janus.js with respect to what it can notify or do).

External projects targeting this, and that we can reference in the resources docs (as others already are) are more than welcome of course!
